src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -436,18 +457,63 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
   private def zipLinearSeq[B](that: collection.LinearSeq[B]): LazyList[(A, B)] =
     newLL {
       if (this.isEmpty || that.isEmpty) State.Empty
-      else sCons((this.head, that.head), tail.zipLinearSeq(that.tail))
+      else sCons((this.head, that.head), this.tail zipLinearSeq that.tail)
     }
 
   private def zipState[B](it: Iterator[B]): State[(A, B)] =
     if (this.isEmpty || !it.hasNext) State.Empty
     else {
       lazy val evaluatedElem = it.next()
-      sCons((head, evaluatedElem), newLL { evaluatedElem; tail.zipState(it) })
+      sCons((head, evaluatedElem), newLL { evaluatedElem; tail zipState it })
     }
 
   override def zipWithIndex: LazyList[(A, Int)] = this.zip(LazyList.from(0))
 
+  override def zipAll[A1 >: A, B](that: scala.Iterable[B], thisElem: A1, thatElem: B): LazyList[(A1, B)] = {
```

Comment:
The type should be `scala.collection.Iterable` instead of `scala.Iterable`

Reply:
does it make a difference? `scala.Iterable` is an alias to `scala.collection.Iterable` (although I did have to look that up to make sure, which isn't necessarily a great sign)